test/ml/algorithms/test_qkernel.py

```diff
@@ -0,0 +1,160 @@
+# This code is part of Qiskit.
```

Comment:
It's good to have end-to-end tests that verify if the whole algorithm runs. But if something goes wrong it can be a little tricky to figure out where things went wrong. Therefore its nice (and less volatile, too) to have more tests that check the individual components, such as circuit construction, raising of errors, initialization and post processing. Could you add some tests that do this, too? (Maybe not all of the above but a subset that makes sense)

Reply:
I've now added many more test cases covering these.